File: features/fintech/saldo_details/src/main/java/com/tokopedia/saldodetails/view/activity/SaldoWebViewActivity.java
package com.tokopedia.saldodetails.view.activity;

import android.content.Context;
import android.content.Intent;
import androidx.fragment.app.Fragment;

import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity;
import com.tokopedia.webview.BaseSessionWebViewFragment;

public class SaldoWebViewActivity extends BaseSimpleActivity {

    private static final String URL = "URL";

    public static Intent getWebViewIntent(Context context, String url) {
        Intent intent = new Intent(context, SaldoWebViewActivity.class);
        intent.putExtra(URL, url);
        return intent;
    }

    @Override
    protected Fragment getNewFragment() {
        return BaseSessionWebViewFragment.newInstance(getIntent().getStringExtra(URL));
    }
}

File: features/fintech/saldo_details/src/main/java/com/tokopedia/saldodetails/view/activity/SaldoWebViewActivity.kt
package com.tokopedia.saldodetails.view.activity

import android.content.Context
import android.content.Intent
import androidx.fragment.app.Fragment

import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity
import com.tokopedia.webview.BaseSessionWebViewFragment

class SaldoWebViewActivity : BaseSimpleActivity() {

    override fun getNewFragment(): Fragment? {
        return BaseSessionWebViewFragment.newInstance(intent.getStringExtra(URL))
    }

    companion object {

        private val URL = "URL"

        fun getWebViewIntent(context: Context, url: String?): Intent {
            val intent = Intent(context, SaldoWebViewActivity::class.java)
            intent.putExtra(URL, url)
            return intent
        }
    }
}
